### ISSUE TITLE
Bun.build fails (and sometimes seg faults) when imports are shared with server (--watch)

### ISSUE THREAD

**Issue by timwhite89** [oven-sh/bun#10813]
### How can we reproduce the crash?

I think I've managed to isolate the minimum to reproduce the errors I've been getting.

This does not seg fault though which is probably because my server project has a more complicated build pipeline that includes svelte + web assembly.

[bun-build-bug.zip](https://github.com/oven-sh/bun/files/15205597/bun-build-bug.zip)
```
> app
│ > app.js
│ > build.js
│ > client0.js
│ > client1.js
│ > client2.js
└ > shared.js
> bug-lib
│ > lib
│ │ > do-test.js
│ └ > index.js
│ > lib2
│ │ > do-other.js
│ └ > index.js
└ > package.json
```

"app" would normally be the web server which builds and serves client js.
"bug-lib" would normally be a game/math library that I am working on.
 
### To reproduce error
1.  set up `bug-lib` for local development
```
/bun-build-bug/>cd bug-lib/
/bun-build-bug/bug-lib/>bun link
bun link v1.1.7-canary.1 (b0b7db5c)
Success! Registered "bug-lib"

To use bug-lib in a project, run:
  bun link bug-lib

Or add it in dependencies in your package.json file:
  "bug-lib": "link:bug-lib"
/bun-build-bug/bug-lib/>cd ..
/bun-build-bug/>cd app
/bun-build-bug/app/>bun add link:bug-lib
bun add v1.1.7-canary.1 (b0b7db5c)

 installed bug-lib@link:bug-lib

 1 package installed [1.00ms]
/bun-build-bug/app/>
```
2. run with `--watch`
```
/bun-build-bug/app/>bun --watch app.js
did the test (shared.js)
did the other test (do-other.js)

building: ./client0.js
 >  /bun-build-bug/app/.dist/client0.js

building: ./client1.js
 >  /bun-build-bug/app/.dist/client1.js

building: ./client2.js
 >  /bun-build-bug/app/.dist/client2.js

building: ./client0.js
 >  /bun-build-bug/app/.dist/client0.js

building: ./client1.js
 >  /bun-build-bug/app/.dist/client1.js

building: ./client2.js
error: Unexpected reading file: "/bun-build-bug/bug-lib/lib2/index.js"
    at /bun-build-bug/bug-lib/lib2/index.js
``` 
### Extra info
`client0.js` uses non library `app/shared.js` which is shared with the server but does not fail
`client1.js` uses `bug-lib` but does not cause a problem because it is not shared with the server
`client2.js` builds fine the first time but fails on subsequent builds

I've been able to share code without error so long as the imports point to different files ie:
```
// bug-lib/package.json
{
  "name": "bug-lib",
  "type": "module",
  "exports": {
    "./lib": "./lib/index.js",
    "./lib2": "./lib2/index.js",
    "./lib2/*": "./lib2/*"
  }
}

// app/app.js
import { T2 } from "bug-lib/lib2"

// app/client2.js
import * as T2 from "bug-lib/lib2/do-other.js"
T2.doOther()
```

### Relevant log output

```shell
============================================================
21 | import { IV2 } from '…/math';
              ^
error: No matching export in "…/index.js" for import "IV2"
    at …/client/entry.svelte:21:10
Bun v1.1.7-canary.1 (b0b7db5c) Linux x64
Args: "bun", "--watch", "server/index.js"
Features: jsc Bun.stderr(2) Bun.stdout(2) abort_signal bunfig define(4) dotenv http_server(2) shell(2) spawn(2) transpiler_cache(4) tsconfig_paths tsconfig(7)
Builtins: "bun:main" "node:events" "node:fs" "node:fs/promises" "node:module" "node:path" "node:stream" "node:string_decoder" "node:timers/promises" "node:util/types"
Elapsed: 4830ms | User: 296ms | Sys: 64ms
RSS: 1.04GB | Peak: 0.25GB | Commit: 1.04GB | Faults: 0

panic: Segmentation fault at address 0x0
oh no: Bun has crashed. This indicates a bug in Bun, not your code.
```

### Stack Trace (bun.report)

Bun v1.1.7 ([`b0b7db5`](https://github.com/oven-sh/bun/tree/b0b7db5c060e830801b0273b7ff6e34e3eb5c254)) on linux x86_64 [AutoCommand]

Segmentation fault at address 0x00000000

- [`ld-temp.o:0`](https://github.com/oven-sh/bun/blob/b0b7db5c060e830801b0273b7ff6e34e3eb5c254/ld-temp.o#L0): `WTF::jscSignalHandler`
- *1 unknown/js code*

**Comment by zackify** [comment]
seeign this as well

**Comment by grmkris** [comment]
seeing here as well

**Comment by timwhite89** [comment]
appears to be fixed now in v1.1.9
